pkg/ratelimiter/interfaces.go

```diff
@@ -0,0 +1,13 @@
+package ratelimiter
+
+import "time"
+
+type RateLimiter interface {
+	// When gets an item and gets to decide how long that item should wait
+	When(item interface{}) time.Duration
+	// Forget indicates that an item is finished being retried.  Doesn't matter whether it's for failing
+	// or for success, we'll stop tracking it
+	Forget(item interface{})
+	// NumRequeues returns back how many failures the item has had
+	NumRequeues(item interface{}) int
+}
```

Comment:
What is the point of duplicating the `controller.RateLimiter` interface? If you want to avoid using the `controller` package prefix, you can just create an alias for it `type RateLimiter = controller.RateLimiter`

Reply:
For now, there is no use for duplicating it. I probably remove it.